src/EnergyPlus/WeatherManager.cc

```diff
@@ -2296,17 +2297,32 @@ namespace WeatherManager {
         state.dataEnvrn->LiquidPrecipitation =
             state.dataWeatherManager->TodayLiquidPrecip(state.dataGlobal->TimeStep, state.dataGlobal->HourOfDay) / 1000.0; // convert from mm to m
         if (state.dataEnvrn->RunPeriodEnvironment) {
-            int EndYear = state.dataEnvrn->EndYear;
-            int CurrentYear = state.dataEnvrn->Year;
-            // only report for the last year
-            if (CurrentYear == EndYear) {
-                int month = state.dataEnvrn->Month;
-                state.dataWaterData->RainFall.MonthlyTotalPrecInWeather[month - 1] += state.dataEnvrn->LiquidPrecipitation * 1000.0;
-                if ((state.dataEnvrn->LiquidPrecipitation > 0) && (state.dataGlobal->TimeStep == 1)) {
-                    state.dataWaterData->RainFall.numRainyHoursInWeather[month - 1] += 1;
-                }
+            int month = state.dataEnvrn->Month;
+            state.dataWaterData->RainFall.MonthlyTotalPrecInWeather[month - 1] += state.dataEnvrn->LiquidPrecipitation * 1000.0;
+            if ((state.dataEnvrn->LiquidPrecipitation > 0) && (state.dataGlobal->TimeStep == 1)) {
+                state.dataWaterData->RainFall.numRainyHoursInWeather[month - 1] += 1;
+            }
+        }
+
+        // if site precipitation not arround, throw warning
+        if (state.dataWaterData->RainFall.ModeID == DataWater::RainfallMode::Invalid) {
```

Comment:
We need to modify this enum for `RainfallMode`. It's currently
```
    enum class RainfallMode
    {
        Invalid = -1,
        RainSchedDesign, // mode of Rainfall determination is Scheduled Design
        IrrSchedDesign,  // mode of Irrigation determination is Scheduled Design
        IrrSmartSched,   // mode of irrigation
        Num
    };
```
I think this needs to be split into two separate enums, one for rainfall, and one for irrigation.  Rainfall should be `None, RainSchedDesign, and EPWPrecipitation` (or something like that).  Then this can be `if ... RainFall.ModeID == DataWater:RainfallMode::EPWPrecipitation` 
* By default, `RainFall.ModeID` should be `None` - meaning nothing in the simulation uses precipitation amounts, so it isn't needed. 
* If Site:Precipitation is present, it should set this to `RainSchedDesign`.  
* If RoofIrrigation has Use Precipitation = Yes and there isn't a Site:Precipitation object, then it should reset this to `EPWPPrecipitation`. 
* If WaterUse:RainCollector is used and there isn't a Site:Precipitation object, then reset this to `EPWPrecipitation`.

The goal here is to avoid doing the precip amount check if it's never going to be used.

Reply:
Agree. I will add an IrrigationMode enum class to hold the irrigation-related settings. I'll also change the flag initialization of the RainfallMode with the logic above.

Reply:
Agreed, I have separated the rain schedule and the irrigation schedule into different enum classes.

I'm unclear about the third bullet.
Precipitation could affect the irrigation in the following ways
1. It directly adds to the water the plants can get
2. It indirectly affects soil moisture, which then affects the irrigation from pipes when the irrigation uses a smart schedule. 
3. It also affects the runoff. 
I thought the "Use irrigation=No" only means in item 1 above, total actual irrigation only counts the water from the pipes. Item 2 and 3 will still consider the impact of rain fall. The flag only resolves how user want to interpret "irrigation". Or should the flag=No mean removing any impact from rain?

Reply:
My understanding is that the current ecoroof model (before your changes) completely ignores rain even from Site:Precipitation.
So the new "Use Rainfall" field will add precipitation plus irrigation to the ecoroof moisture balance when =Yes, and ignore rain when = No.